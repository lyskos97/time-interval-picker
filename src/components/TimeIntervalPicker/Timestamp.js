/* @flow */
/* eslint-disabl */

import * as React from 'react';
import fns from 'date-fns';
import { type Stamp } from './TimeIntervalPicker';
import s from './TimeIntervalPicker.scss';

type Props = {
  handleClick?: () => void,
} & Stamp;

export default class Timestamp extends React.Component<Props, void> {
  formatTime(d: Date): string {
    return fns.format(d, 'HH:mm');
  }

  handleClick = () => {
    const { handleClick } = this.props;

    if (handleClick) {
      handleClick();
    }
  };

  render() {
    const { value, status } = this.props;
    const classnames = [s.cell];
    let onClick = this.handleClick;

    if (status === 'selected') classnames.push(s.cellSelected);
    else if (status === 'disabled' || status === 'reserved') {
      classnames.push(s.cellInactive);
      onClick = null;
    } else classnames.push(s.cellActive);

    return (
      <div className={classnames.join(' ')} onClick={onClick}>
        <span className="time">{this.formatTime(value)}</span>
      </div>
    );
  }
}
