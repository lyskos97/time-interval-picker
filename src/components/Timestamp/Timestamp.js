/* @flow */
/* eslint-disable */

import * as React from 'react';
import fns from 'date-fns';

type Props = {
  time: Date,
  status: 'selected' | 'active' | 'inactive',
};

export default class Timestamp extends React.Component<Props, void> {
  formatTime(d: Date): string {
    return fns.format(d, 'HH:mm');
  }

  render() {
    const { time, status } = this.props;
    let classname = 'cell';

    console.log(time.toLocaleTimeString(), status);

    if (status === 'selected') classname += ' cellSelected';
    else if (status === 'inactive') classname += ' cellInactive';
    else classname += ' cellActive';

    return (
      <div className={classname}>
        <span className="time">{this.formatTime(time)}</span>
      </div>
    );
  }
}
