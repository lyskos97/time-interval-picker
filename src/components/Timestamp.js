/* @flow */
/* eslint-disabl */

import * as React from 'react';
import fns from 'date-fns';
import { type Stamp } from './TimeIntervalPicker';

type Props = {
  handleClick?: () => void,
} & Stamp;

export default class Timestamp extends React.Component<Props, void> {
  formatTime(d: Date): string {
    return fns.format(d, 'HH:mm');
  }

  handleClick = () => {
    const { handleClick } = this.props;

    if (handleClick) {
      handleClick();
    }
  };

  render() {
    const { value, status } = this.props;
    let classname = 'cell';

    if (status === 'selected') classname += ' cellSelected';
    else if (status === 'disabled') classname += ' cellInactive';
    else classname += ' cellActive';

    return (
      <div className={classname} onClick={this.handleClick}>
        <span className="time">{this.formatTime(value)}</span>
      </div>
    );
  }
}
