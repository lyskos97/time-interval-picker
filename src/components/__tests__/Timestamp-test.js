/* @flow */

import * as React from 'react';
import Timestamp from '../Timestamp';

describe('Timestamp', () => {
  describe('render()', () => {
    it('basic', () => {
      const wrapper = shallow(<Timestamp value={new Date(2018, 0)} />);

      expect(wrapper).toMatchSnapshot();
    });

    it('status `selected`', () => {
      const wrapper = shallow(<Timestamp value={new Date(2018, 0)} status="selected" />);

      expect(wrapper).toMatchSnapshot();
    });

    it('status `disabled`', () => {
      const wrapper = shallow(<Timestamp value={new Date(2018, 0)} status="disabled" />);

      expect(wrapper).toMatchSnapshot();
    });

    it('status `available`', () => {
      const wrapper = shallow(<Timestamp value={new Date(2018, 0)} status="available" />);

      expect(wrapper).toMatchSnapshot();
    });
  });

  it('formatTime()', () => {
    const time = new Date(2018, 0, 1, 13, 37);
    const wrapper = shallow(<Timestamp value={time} />);
    const formattedTime = wrapper.instance().formatTime(time);

    expect(formattedTime).toBe('13:37');
  });

  it('handleClick()', () => {
    const handleClick = jest.fn();
    const wrapper = shallow(<Timestamp value={new Date(2018, 0)} handleClick={handleClick} />);
    wrapper.find('div').simulate('click');

    expect(handleClick).toBeCalled();
  });
});
