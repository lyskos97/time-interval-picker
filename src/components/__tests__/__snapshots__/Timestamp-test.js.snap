// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timestamp render() basic 1`] = `
<div
  className="cell cellActive"
  onClick={[Function]}
>
  <span
    className="time"
  >
    00:00
  </span>
</div>
`;

exports[`Timestamp render() status \`active\` 1`] = `
<div
  className="cell cellActive"
  onClick={[Function]}
>
  <span
    className="time"
  >
    00:00
  </span>
</div>
`;

exports[`Timestamp render() status \`inactive\` 1`] = `
<div
  className="cell cellInactive"
  onClick={[Function]}
>
  <span
    className="time"
  >
    00:00
  </span>
</div>
`;

exports[`Timestamp render() status \`selected\` 1`] = `
<div
  className="cell cellSelected"
  onClick={[Function]}
>
  <span
    className="time"
  >
    00:00
  </span>
</div>
`;
